[Issue]: Building MiOpen

### Problem Description

```
(.venv) C:\Users\corerole\source\repos\TheRock\build>..\..\cmake-3.31.11-windows-x86_64\bin\cmake.exe --build . --target therock-dist --config Release --parallel 16
[0/2] Re-checking globbed directories...
[1/4] Building sub-project MIOpen (in background)
FAILED: [code=1] ml-libs/MIOpen/stamp/build.stamp C:/Users/corerole/source/repos/TheRock/build/ml-libs/MIOpen/stamp/build.stamp
C:\Windows\system32\cmd.exe /C "cd /D C:\Users\corerole\source\repos\TheRock\build\ml-libs\MIOpen\build && C:\Users\corerole\source\repos\TheRock\.venv\Scripts\python3.exe C:/Users/corerole/source/repos/TheRock/build_tools/teatime.py --log-timestamps --label MIOpen --interactive C:/Users/corerole/source/repos/TheRock/build/logs/MIOpen_build.log -- C:/Users/corerole/source/repos/cmake-3.31.11-windows-x86_64/bin/cmake.exe -E env --unset=ROCM_PATH --unset=ROCM_DIR --unset=HIP_PATH --unset=HIP_DIR -- C:/Users/corerole/source/repos/cmake-3.31.11-windows-x86_64/bin/cmake.exe --build C:/Users/corerole/source/repos/TheRock/build/ml-libs/MIOpen/build && C:\Users\corerole\source\repos\cmake-3.31.11-windows-x86_64\bin\cmake.exe -E touch C:/Users/corerole/source/repos/TheRock/build/ml-libs/MIOpen/stamp/build.stamp"
[MIOpen] [0/2] Re-checking globbed directories...
[MIOpen] [1/360] C:\Windows\system32\cmd.exe /C "cd /D C:\Users\corerole\source\repos\TheRock\build\ml-libs\MIOpen\build && C:\Users\corerole\source\repos\cmake-3.31.11-windows-x86_64\bin\cmake.exe -E rm -f C:/Users/corerole/source/repos/TheRock/build/ml-libs/MIOpen/stamp/stage.stamp"
[MIOpen] [2/360] Inlining kernels batch #96
[MIOpen] FAILED: [code=1] inlined_kernels/batch_96.cpp.hpp C:/Users/corerole/source/repos/TheRock/build/ml-libs/MIOpen/build/inlined_kernels/batch_96.cpp.hpp
[MIOpen] src\CMakeFiles\batch_96.cpp.hpp-d9b5a80.bat e3b8456fd79cae84
[MIOpen] File not found: "C:/Users/corerole/source/repos/TheRock/rocm-libraries/projects/miopen/src/kernels/dynamic_igemm/igemm_gtc_xdlops_nhwc_gfx950/bwd_fp16/igemm_bwd_gtcx35_nhwc_fp16_bx0_ex1_bt128x128x16_wt32x32x4_ws1x1_wr2x2_ta1x1x8x1_1x16x1x16_tb1x1x8x1_1x16x1x16_mh_me_vs1_gkgs.s"
[MIOpen] Batch file failed at line 3 with errorcode 1
[MIOpen] ninja: build stopped: subcommand failed.
[MIOpen FAILED WITH CODE 1 in 0 seconds]
ninja: build stopped: subcommand failed.
```

### Operating System

Windows 11Pro

### CPU

Ryzen AI 350

### GPU

860M

### ROCm Version

trunc

### ROCm Component

MIOpen

### Steps to Reproduce

cmake.exe -GNinja .. -DTHEROCK_AMDGPU_DIST_BUNDLE_NAME=custom -DTHEROCK_AMDGPU_FAMILIES="gfx1152" -DCMAKE_BUILD_TYPE=Release  -DTHEROCK_ENABLE_MIOPEN_PLUGIN=OFF -DTHEROCK_ENABLE_HIPDNN_SAMPLES=OFF

cmake.exe --build . --target therock-dist --config Release --parallel 16


### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

The problem is with the addkernels.exe utility.
Since it lists files with multiple characters in their names with a single command from the console, it causes a terminal error.

Please consider adding a function to read the file list from a file.

(c) Google Translate

--- a/projects/miopen/addkernels/addkernels.cpp
+++ b/projects/miopen/addkernels/addkernels.cpp
@@ -1,28 +1,6 @@
-/*******************************************************************************
- *
- * MIT License
- *
- * Copyright (c) 2017 Advanced Micro Devices, Inc.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
- *
- *******************************************************************************/
+// Copyright © Advanced Micro Devices, Inc., or its affiliates.
+// SPDX-License-Identifier:  MIT
+
 #include "include_inliner.hpp"
 #include "miopen/filesystem.hpp"
 #include <algorithm>
@@ -316,7 +294,27 @@ int main(int argc, char* argv[])
         std::string arg(argv[i]);
         std::transform(arg.begin(), arg.end(), arg.begin(), ::tolower);
 
-        if(arg == "-s" || arg == "-source")
+        // Handle response file (starts with @)
+        if(arg.starts_with('@'))
+        {
+            std::string response_file = std::string(argv[i]).substr(1);
+            std::ifstream file(response_file);
+            if(!file.is_open())
+            {
+                std::cerr << "Error: Cannot open response file: " << response_file << std::endl;
+                return 1;
+            }
+
+            std::string line;
+            while(std::getline(file, line))
+            {
+                if(!line.empty() && !line.starts_with('#'))
+                {
+                    sourceFiles.emplace_back(line);
+                }
+            }
+        }
+        else if(arg == "-s" || arg == "-source")
         {
             while(++i < argc && *argv[i] != '-')
                 sourceFiles.emplace_back(argv[i]);

--- a/projects/miopen/src/CMakeLists.txt
+++ b/projects/miopen/src/CMakeLists.txt
@@ -813,12 +813,19 @@ if( MIOPEN_BACKEND MATCHES "OpenCL" OR MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN
                 set(KERNEL_SRC_ASM_PATH ${PROJECT_BINARY_DIR}/inlined_kernels/batch_${KERNELS_BATCH_ID}.S)
                 set(KERNEL_SRC_BIN_PATH ${PROJECT_BINARY_DIR}/inlined_kernels/batch_${KERNELS_BATCH_ID}.bin)
 
+                # Generate response file for this batch to avoid command line length limits
+                set(KERNEL_RESPONSE_FILE ${PROJECT_BINARY_DIR}/inlined_kernels/batch_${KERNELS_BATCH_ID}_files.txt)
+                file(WRITE ${KERNEL_RESPONSE_FILE} "")
+                foreach(kernel_file ${KERNELS_BATCH})
+                    file(APPEND ${KERNEL_RESPONSE_FILE} "${kernel_file}\n")
+                endforeach()
+
                 if(MIOPEN_INCBIN)
                     add_custom_command(
                         OUTPUT ${KERNEL_SRC_HPP_PATH} ${KERNEL_SRC_ASM_PATH} ${KERNEL_SRC_BIN_PATH}
                         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                        DEPENDS addkernels ${KERNELS_BATCH} ${KERNEL_INCLUDES}
-                        COMMAND $<TARGET_FILE:addkernels> -asm ${KERNEL_SRC_ASM_PATH} ${KERNEL_SRC_BIN_PATH} -target ${KERNEL_SRC_HPP_PATH} -extern ${EXTRA_OPTIONS} -source ${KERNELS_BATCH}
+                        DEPENDS addkernels ${KERNELS_BATCH} ${KERNEL_INCLUDES} ${KERNEL_RESPONSE_FILE}
+                        COMMAND $<TARGET_FILE:addkernels> -asm ${KERNEL_SRC_ASM_PATH} ${KERNEL_SRC_BIN_PATH} -target ${KERNEL_SRC_HPP_PATH} -extern ${EXTRA_OPTIONS} @${KERNEL_RESPONSE_FILE}
                         COMMENT "Inlining kernels batch #${KERNELS_BATCH_ID}${MESSAGE_SUFFIX}"
                         )
                     list(APPEND MIOpen_Source ${KERNEL_SRC_HPP_PATH} ${KERNEL_SRC_ASM_PATH})
@@ -826,8 +833,8 @@ if( MIOPEN_BACKEND MATCHES "OpenCL" OR MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN
                     add_custom_command(
                         OUTPUT ${KERNEL_SRC_HPP_PATH}
                         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                        DEPENDS addkernels ${KERNELS_BATCH} ${KERNEL_INCLUDES}
-                        COMMAND $<TARGET_FILE:addkernels> -target ${KERNEL_SRC_HPP_PATH} -extern ${EXTRA_OPTIONS} -source ${KERNELS_BATCH}
+                        DEPENDS addkernels ${KERNELS_BATCH} ${KERNEL_INCLUDES} ${KERNEL_RESPONSE_FILE}
+                        COMMAND $<TARGET_FILE:addkernels> -target ${KERNEL_SRC_HPP_PATH} -extern ${EXTRA_OPTIONS} @${KERNEL_RESPONSE_FILE}
                         COMMENT "Inlining kernels batch #${KERNELS_BATCH_ID}${MESSAGE_SUFFIX}"
                         )
                     configure_file(kernels/kernels_batch.cpp.in ${KERNEL_SRC_CPP_PATH})

